From: Alex Gonzalez <alexg@balena.io>
Date: Fri, 17 Jun 2022 14:50:13 +0200
Subject: [PATCH] common.mk: Add current directory as git safe directory

The security vulnerability CVE-2022-24765 was fixed in git v2.35.2 enforces
new access controls. This has been addressed in newer Poky versions with
the addition of a git-intercept script.
(see 4d7383aefb391a5a998454c70feb96127951ca0a)

Until then, patch the build scripts to configure the working directory
as safe.

Upstream-Status: Inappropriate (upstream uses git intercept)
Signed-off-by: Alex Gonzalez <alexg@balena.io>
---
 mk/common.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mk/common.mk b/mk/common.mk
index f8170de83a66..346b09da03b9 100644
--- a/mk/common.mk
+++ b/mk/common.mk
@@ -21,7 +21,7 @@ DOCKER   ?= docker
 UID      := $(shell id -u)
 GID      := $(shell id -g)
 DATE     := $(shell date -u --iso-8601=minutes)
-REVISION := $(shell git rev-parse HEAD)
+REVISION := $(shell git config --global --add safe.directory ${PWD} && git rev-parse HEAD)
 COMPILER := $(realpath $(shell which $(firstword $(CC))))
 PLATFORM ?= $(shell uname -m)
 JETSON   ?= TRUE
